hyperspy/signal.py

```diff
@@ -3936,12 +3934,15 @@ def _iterate_signal(self):
             getitem[unfolded_axis] = i
             yield(data[getitem])
 
-    def _remove_axis(self, axis):
+    def _remove_axes(self, axes):
         am = self.axes_manager
-        axis = am[axis]
-        if am.navigation_dimension + am.signal_dimension > 1:
-            am.remove(axis.index_in_axes_manager)
-            if axis.navigate is False:  # The removed axis is a signal axis
+        axes = am[axes]
+        if not np.iterable(axes):
+            axes = (axes,)
+        if am.navigation_dimension + am.signal_dimension > len(axes):
```

Comment:
Or maybe that can be confused with iteration length (since AxesManager iterates of navigation axes)..